### DISCUSSION TITLE
[HELP] 401 error  code using functions

### DISCUSSION THREAD

**Discussion by Kappyh** [appwrite/appwrite#3154] [Q&A]
Hi guys

I have a question related to the appwrite functions, I made one function following the [example in the tutorial](https://appwrite.io/docs/functions), I deployed the function and running in the appwrite console works just fine.
But I did a web app to test the function, installed the sdk and called the function... but always shows a message with the 401 code, saying the guest user doesnt have write permitions.
I return to the appwrite panel and checked all the permissions for the function, but still showing that error code with the same write permission problem for the **guest user.**

```
{message: "User (role: guest) missing scope (execution.write)", code: 401,…}
code: 401
message: "User (role: guest) missing scope (execution.write)"
type: "general_unauthorized_scope"
version: "0.13.4
```

I looked in the example in the docs and I didn't found the place where I can indicate a user for the function call in the SDK.

I'm stuck in this problem for 5 days, someone know how to solve or can give me some direction?

**Comment by christyjacob4** [comment]
Hi @Kappyh , Sorry for the delay. Can you share a screenshot of your function settings and the code you're using to trigger the function?

**Reply by Kappyh** [reply]
No problem @christyjacob4 .
Yes, sure. This is my code:

My html code:
![html_code](https://popcoins.dev/public_images/example_html.png)

My JS code:
![js_code](https://popcoins.dev/public_images/example_js.png)

Permissions on my function:
https://popcoins.dev/public_images/permissions.gif

**Comment by Kappyh** [comment]
@christyjacob4 Is something else I can provide for you?
Also, I'm using this appwrite application on digital ocean.

**Reply by gewenyu99** [reply]
Hi there,

I think you're missing a user session for your Appwrite function with the appropriate permissions.

There are two things that you'll need to do

1. Go to the settings tab of your function in Console and set `Execution Access` for the function. If you would like everyone to access this function, even guests, use `role:all`, but I strongly recommend you read the [permission docs](https://appwrite.io/docs/permissions) to find the most specific permission appropriate for your use case.

2. Before executing the function, you need to actually log in to a user, so your Appwrite SDK has an active session. You can create an account session [like this](https://appwrite.io/docs/client/account?sdk=web-default#accountCreateSession). You may also need the user to first sign up, which you can do (like this)[https://appwrite.io/docs/client/account?sdk=web-default#accountCreate].
